src/Microsoft.Diagnostics.Runtime/src/DataReaders/MacOS/MacOSProcessDataReader.cs

```diff
@@ -0,0 +1,364 @@
+
+        public void FlushCachedData()
+        {
+            _memoryRegions = LoadMemoryRegions();
+        }
+
+        public IEnumerable<ModuleInfo> EnumerateModules()
+        {
+            int taskInfoCount = Native.TASK_DYLD_INFO_COUNT;
+            int kr = Native.task_info(_task, Native.TASK_DYLD_INFO, out Native.task_dyld_info dyldInfo, ref taskInfoCount);
+            if (kr != 0)
+                throw new ClrDiagnosticsException();
+
+            Native.dyld_all_image_infos infos = Read<Native.dyld_all_image_infos>(dyldInfo.all_image_info_addr);
+            for (uint i = 0; i < infos.infoArrayCount; i++)
+            {
+                Native.dyld_image_info info = Read<Native.dyld_image_info>(infos.infoArray, i);
+                ulong imageAddress = info.imageLoadAddress;
+                string imageFilePath = ReadNullTerminatedAscii(info.imageFilePath);
+                yield return new ModuleInfo(imageAddress, imageFilePath, true, 0, 0, ImmutableArray<byte>.Empty);
+            }
+
+            unsafe T Read<T>(ulong address, uint index = 0)
+                where T : unmanaged
+            {
+                T result;
+                if (Native.vm_read_overwrite(_task, address + index * (uint)sizeof(T), sizeof(T), &result, out _) != 0)
+                    throw new ClrDiagnosticsException();
+
+                return result;
+            }
+
+            unsafe string ReadNullTerminatedAscii(ulong address)
+            {
+                StringBuilder builder = new StringBuilder(64);
+                byte* bytes = stackalloc byte[64];
+
+                bool done = false;
+                while (!done && (Native.vm_read_overwrite(_task, address, 64, bytes, out long read) == 0))
+                {
+                    address += (ulong)read;
+                    for (int i = 0; !done && i < read; i++)
+                    {
+                        if (bytes[i] != 0)
+                            builder.Append((char)bytes[i]);
+                        else
+                            done = true;
+                    }
+                }
+
+                return builder.ToString();
+            }
+        }
+
+        public ImmutableArray<byte> GetBuildId(ulong baseAddress) => ImmutableArray<byte>.Empty;
+
+        public bool GetVersionInfo(ulong baseAddress, out VersionInfo version)
+        {
+            // TODO
+            version = default;
+            return false;
+        }
+
+        public override unsafe int Read(ulong address, Span<byte> buffer)
+        {
+            DebugOnly.Assert(!buffer.IsEmpty);
+
+            int readable = GetReadableBytesCount(address, buffer.Length);
+            if (readable <= 0)
+            {
+                return 0;
+            }
+
+            fixed (byte* ptr = buffer)
+            {
+                int kr = Native.vm_read_overwrite(_task, address, readable, ptr, out long read);
+                if (kr != 0)
+                    throw new ClrDiagnosticsException();
```

Comment:
```suggestion
                    return 0;
```

Similar to above, let's just return 0 instead of throwing if we can't read from the process.